src/main/java/com/minecolonies/util/BlockPosUtil.java

```diff
@@ -144,10 +152,22 @@ public static boolean isClose(@NotNull BlockPos coordinate, @NotNull EntityCitiz
      */
     public static long getDistanceSquared(@NotNull BlockPos block1, @NotNull BlockPos block2)
     {
-        long i = (long) block1.getX() - block2.getX();
-        long i1 = (long) block1.getY() - block2.getY();
-        long i2 = (long) block1.getZ() - block2.getZ();
-        return i * i + i1 * i1 + i2 * i2;
+        final long xDiff = Math.abs((long) block1.getX() - block2.getX());
+        final long yDiff = Math.abs((long) block1.getY() - block2.getY());
+        final long zDiff = Math.abs((long) block1.getZ() - block2.getZ());
+
+        if (xDiff > MAX_SQRT || yDiff > MAX_SQRT || zDiff > MAX_SQRT)
```

Comment:
This whole check is unnecessary. I did some simple math, even if minecraft worlds were 10 times bigger than they are currently a long would still work for us.